documentation/modules/auxiliary/scanner/ssh/ssh_login_pubkey.md

```diff
@@ -31,6 +33,10 @@
 
   A string to the private key to attempt, or a folder containing private keys to attempt.  Any file name starting with a period (`.`) or ending in `.pub` will be ignored.
   An SSH key is typically kept in a user's home directory under `.ssh/id_rsa`.  The file contents, when not encrypted with a password will start with `-----BEGIN RSA PRIVATE KEY-----`
+
+  **PRIVATE_KEY**
+
+  A string of the private key to attempt. For MSFConsole users the option should be set to `file:PRIVATE_KEY_PATH` and it will read in the string value of the private key. Currently OpenSSH, RSA, DSA, and EC private keys are supported. 
   
```

Comment:
I tried a bunch of EC private keys, and wasn't able to get it working:

```
$ openssl ecparam -genkey -name prime256v1 -noout -out ec256-key-pair.pem
$ cat ec256-key-pair.pem 
-----BEGIN EC PRIVATE KEY-----
MHcCAQEEINaA36xSxGFfmZd5vVC9D86jdKyFJ7Ac4pZ+1eP8UCwEoAoGCCqGSM49
AwEHoUQDQgAEYdb2Zykq+ks3sx85vRffmydSFICHWpj/OMjQ8HHGqWG4zlUufQwF
BfXIsY0QpRS8wyI+ZjIvJsLOLu4tRoZEqg==
-----END EC PRIVATE KEY-----

```
Then in MSF:
```
msf6 auxiliary(scanner/ssh/ssh_login_pubkey) > set private_key file:/testing/ec256-key-pair.pem
private_key => -----BEGIN EC PRIVATE KEY-----
MHcCAQEEINaA36xSxGFfmZd5vVC9D86jdKyFJ7Ac4pZ+1eP8UCwEoAoGCCqGSM49
AwEHoUQDQgAEYdb2Zykq+ks3sx85vRffmydSFICHWpj/OMjQ8HHGqWG4zlUufQwF
BfXIsY0QpRS8wyI+ZjIvJsLOLu4tRoZEqg==
-----END EC PRIVATE KEY-----

msf6 auxiliary(scanner/ssh/ssh_login_pubkey) > run

[*] 139.59.231.55:22 SSH - Testing Cleartext Keys
[-] Auxiliary failed: RuntimeError Invalid private key
[-] Call stack:
[-]   /home/smash/git/metasploit-framework/modules/auxiliary/scanner/ssh/ssh_login_pubkey.rb:257:in `valid?'
[-]   /home/smash/git/metasploit-framework/modules/auxiliary/scanner/ssh/ssh_login_pubkey.rb:128:in `run_host'
[-]   /home/smash/git/metasploit-framework/lib/msf/core/auxiliary/scanner.rb:121:in `block (2 levels) in run'
[-]   /home/smash/git/metasploit-framework/lib/msf/core/thread_manager.rb:105:in `block in spawn'
[*] Auxiliary module execution completed
```

Reply:
PEM Key format for SSH usage is very specific. The format requires: 
```
-----BEGIN OPENSSH PRIVATE KEY-----
Key material
-----END OPENSSH PRIVATE KEY-----
```

Reply:
Testing with keys generated using `ssh-keygen` may produce different results. Valid types would be ` [-t dsa | ecdsa | ed25519 | rsa]`.  I will verify these types on released versions of framework as well.

Reply:
Yeah it appears the validator within `ssh_login_pubkey` only validates for keys that match `!!(key_data.match(/BEGIN [RECD]SA PRIVATE KEY/) && !key_data.match(/Proc-Type:.*ENCRYPTED/))`. I'll need to clarify that not all EC keys are supported.